src/test/ui/issues/issue-65611.stderr

```diff
@@ -2,7 +2,10 @@ error[E0282]: type annotations needed
   --> $DIR/issue-65611.rs:59:20
    |
 LL |     let x = buffer.last().unwrap().0.clone();
-   |                    ^^^^ cannot infer type for `T`
+   |                    ^^^^
+   |                    |
+   |                    cannot infer type for `T`
```

Comment:
Pre-existing, but it's not great that we say "cannot infer type for `T`" here -- I at least am often puzzled by this message. ("What is this T??")

I wonder if we could detect that it's the return value and say something like:

> cannot infer return value from this call from context

Reply:
I think this would be less confusing if we pointed at the definition of `last` so we display, in context, that it returns `Option<T>`, for example.

Reply:
Yes, that would help a lot

Reply:
I couldn't manage to point at the original type, but this issue is now reduced or eliminated with the new output (when we can suggest a type param, it makes clear what `T` is, and when we don't we show the full return type that includes `T`).